packages/cli/src/decorators/registerController.ts

```diff
@@ -0,0 +1,36 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { Router } from 'express';
+import type { Application, Request, RequestHandler as Middleware, Response } from 'express';
+import { BASE_PATH, Controller, RouteMetadata, ROUTES } from './types';
+import { send } from '@/ResponseHelper'; // TODO: move `ResponseHelper.send` to this file
+import type { Config } from '@/config';
+
+export const registerController = (
+	app: Application,
+	config: Config,
+	instance: object,
+	...middlewares: Middleware[]
+): void => {
+	const controllerClass = instance.constructor;
+	const routes = Reflect.getMetadata(ROUTES, controllerClass) as RouteMetadata[];
+	if (routes.length > 0) {
+		const router = Router({ mergeParams: true });
+		const restEndpoint = config.getEnv('endpoints.rest');
+		const basePath = Reflect.getMetadata(BASE_PATH, controllerClass) as string;
```

Comment:
1. yeah. changed to `restBasePath` and `controllerBasePath`
2. `getMetadata` could return undefined if someone calls `registerController` on a controller class that's missing the decorator. updated this to simply crash the server. something like this would be caught easily in dev/test